src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/VMDynamicTests.ps1

```diff
@@ -0,0 +1,341 @@
+﻿# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+function get_all_vm_locations
+{
+    $locations = Get-AzureLocation | where { $_.Name -like 'Microsoft.Compute/virtualMachines' } | select -ExpandProperty Locations;
+
+    for ($i = 0; $i -lt $locations.Count; $i++)
+    {
+        $locations[$i] = ($locations[$i] -Replace ' ', '').ToLower();
```

Comment:
Well, I don't think all users must use the return from Get-AzureLocation, and for example, I may simply know that ```westus``` works, but I can understand your suggestion -- you would like to validate that the return values from ```Get-AzureLocation``` can be used here.

I would argue that the dynamic tests' focus is not yet about different forms of location strings (for which we have different test cases), as VM create/remove scenario is.

I would suggest we create a new item for follow-up.